### PR TITLE
Bump ESLint, alphabetize es6 features, add modules, remove redundant no-comma dangle

### PR DESCRIPTION
**Pull request by rtsao**
`no-comma-dangle` is replaced with `comma-dangle` which has been set, so I removed the redundant rule.

### CODE REVIEW

File: rc/.eslintrc.es6.json [hunk 1]
```diff
@@ -59,23 +59,25 @@
...
         "objectLiteralShorthandProperties": true,
-        "objectLiteralDuplicateProperties": true,
-        "generators": true,
-        "destructuring": true,
-        "classes": true
+        "octalLiterals": true,
+        "regexUFlag": true,
+        "regexYFlag": true,
+        "restParams": true,
+        "spread": true,
```
Comment (rtsao) [inline rc/.eslintrc.es6.json:88]: The additions of `spread` and `modules` are the only changes besides alphabetization.

**Comment by andrewdeandrade** [comment]
changes:
- indent: add reasonable defaults for the `SwitchCase` and `VariableDeclarator` options
- block-spacing: [2, "always"]
- callback-return: [2, ["callback", "cb", "next", "done"]]
- no-invalid-this: 2
- no-implicit-coercion: 2
- no-useless-call: 2
- id-length: [0, {"min": 1, "max": 80, "properties": "always", "exceptions": []}]
- init-declarations: 0
- id-match: 0 

es6
- newTarget: should this be true?
- no-const-assign: should there be a rule for this?
- arrow-parens: add a reasonable default for this missing rule configuration
- arrow-spacing: add a reasonable default for this missing rule configuration
- prefer-arrow-callback: add a reasonable default for this missing rule configuration
- prefer-const: add a reasonable default for this missing rule configuration
- prefer-spread: add a reasonable default for this missing rule configuration
- prefer-reflect: add a reasonable default for this missing rule configuration
- prefer-template: add a reasonable default for this missing rule configuration
- require-yield: add a reasonable default for this missing rule configuration
- no-dupe-class-members: add a reasonable default for this missing rule configuration
- no-class-assign: add a reasonable default for this missing rule configuration

Err on the side of being more restrictive and don't worry about a major version bump with these changes.

**Comment by andrewdeandrade** [comment]
Is this ready to merge?

**Comment by rtsao** [comment]
Yes, should be good to go.

**Pull request merged**